SimpleWAWS/Code/CsmExtensions/CsmSubscriptionExtensions.cs

```diff
@@ -13,39 +13,101 @@ namespace SimpleWAWS.Code.CsmExtensions
+                                    && ((r.tags.ContainsKey("Bad") || !r.tags.ContainsKey("FunctionsContainerDeployed"))
+                                        && (!r.tags.ContainsKey("UserId"))
+                                        && r.properties.provisioningState != "Deleting"))
+                        .Select(async r => await
+                            Delete(
+                                await
+                                    Load(new ResourceGroup(subscription.SubscriptionId, r.name), r,
+                                        loadSubResources: false), block: false));
+              
+                    await deleteBadResourceGroupsTasks.IgnoreFailures().WhenAll();
+
+                    //reload after deleting the bad subs
+                    //TODO: Ensure a background task always takes care of this  
+                    csmResourceGroups = await subscription.LoadResourceGroupsForSubscription();
+                }
+                var csmSubscriptionResourcesReponse =
+                    await csmClient.HttpInvoke(HttpMethod.Get, ArmUriTemplates.SubscriptionResources.Bind(subscription));
+
+                await csmSubscriptionResourcesReponse.EnsureSuccessStatusCodeWithFullError();
+
+                var csmSubscriptionResources =
+                    await csmSubscriptionResourcesReponse.Content.ReadAsAsync<CsmArrayWrapper<object>>();
+
+                var goodResourceGroups = csmResourceGroups.value
+                    .Where(r => IsSimpleWaws(r))
+                    .Select(r => new
+                    {
+                        ResourceGroup = r,
+                        Resources =
+                            csmSubscriptionResources.value.Where(
+                                resource => resource.id.IndexOf(r.id, StringComparison.OrdinalIgnoreCase) != -1)
+                    });
+
+                subscription.ResourceGroups = await goodResourceGroups
+                    .Select(
+                        async r =>
+                            await
+                                Load(new ResourceGroup(subscription.SubscriptionId, r.ResourceGroup.name),
+                                    r.ResourceGroup, r.Resources))
+                    .IgnoreAndFilterFailures();
+
+                return subscription;
+            }
+            else
+            {
+                return await LoadJenkinsSubscription(subscription);
+            }
+        }
 
-            //Make sure to register for AppServices RP at least once for each sub
-            await csmClient.HttpInvoke(HttpMethod.Post, ArmUriTemplates.WebsitesRegister.Bind(subscription));
-            await csmClient.HttpInvoke(HttpMethod.Post, ArmUriTemplates.AppServiceRegister.Bind(subscription));
-            await csmClient.HttpInvoke(HttpMethod.Post, ArmUriTemplates.StorageRegister.Bind(subscription));
-            
+        private static async Task<Subscription> LoadJenkinsSubscription(Subscription subscription)
```

Comment:
as far as I can tell this doesn't seem all that different from the normal load sub function, can't they be refactored into 1 function?